src/lib/scanner.ts

```diff
@@ -39,53 +50,46 @@ export class ScanRunner {
      */
     public async runDfa(targets: string[], projectDir: string): Promise<string> {
         // Create the arg array.
-        const args: string[] = this.createDfaArgArray(targets, projectDir);
+        const args: Inputs = this.createDfaArgArray(targets, projectDir);
 
         // Invoke the scanner.
-        const executionResult: ExecutionResult = await this.invokeAnalyzer(args);
+        const executionResult: AnyJson = await this.invokeAnalyzerWithoutCli(args);
 
         // Process the results.
-        return this.processDfaResults(executionResult);
+        return executionResult as string;
     }
 
     /**
      * Creates the arguments for an execution of {@code sf scanner run dfa}, for use in a child process.
      * @param targets The files/methods to be targeted.
      * @param projectDir The root of the project to be scanned.
      */
-    private createDfaArgArray(targets: string[], projectDir: string): string[] {
-        const args: string[] = [
-            'scanner', 'run', 'dfa',
-            '--target', `${targets.join(',')}`,
-            `--projectdir`, projectDir,
-            // NOTE: For now, we're using HTML output since it's the easiest to display to the user.
-            //       This is exceedingly likely to change as we refine and polish the extension.
-            `--format`, `html`,
-            // NOTE: Using `--json` gives us easily-processed results, but denies us access to some
-            //       elements of logging, most notably the informative progress reporting provided
-            //       by the engine's spinner. This was deemed an acceptable trade-off during initial
-            //       implementation, but we may wish to rethink this in the future as we polish things.
-            `--json`
-        ];
+    private createDfaArgArray(targets: string[], projectDir: string): Inputs {
+        const args: Inputs = {
+            'targets': `${targets.join(',')}`,
+            'projectdir': [projectDir],
+            'format': `html`,
```

Comment:
Why do we format as html for runDfa but json for run? That seems odd.

Reply:
Because DFA violations have both a source and a sink, whereas non-DFA violations just have one location, it wasn't immediately clear how to actually display information about DFA violations inline.
So we just have the scanner return results in its existing HTML format, and then we display that content in a new tab.